data-prepper-api/src/main/java/org/opensearch/dataprepper/model/failures/DlqObject.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.model.failures;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Objects;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A model representing DLQ objects in Data Prepper
+ *
+ * @since 2.2
+ */
+public class DlqObject {
+
+    private static final String ISO8601_FORMAT_STRING = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'";
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern(ISO8601_FORMAT_STRING)
+            .withZone(ZoneId.systemDefault());;
+
+    private static final String VERSION = "1";
+
+    private final String pluginId;
+
+    private final String pluginName;
+
+    private final String pipelineName;
+
+    private final String version;
+
+    private final Object failedData;
+
+    private final String timestamp;
```

Comment:
You could keep this as an `Instant` and let Jackson handle serialization. But, it's internal, so I'm ok either way.